ra_aid/agent_utils.py

```
@@ -56,58 +64,202 @@
+            logger.debug(f"Could not find token limit for {provider}/{model_name}")
+
+        return max_input_tokens
+
+    except Exception as e:
+        logger.warning(f"Failed to get model token limit: {e}")
+        return None
+
+
+def build_agent_kwargs(
+    checkpointer: Optional[Any] = None,
+    config: Dict[str, Any] = None,
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)